buckets/redis/bucket.go

```diff
@@ -50,31 +50,25 @@ func (a *abstractBucket) Take(requested int64, maxWaitTime time.Duration) (time.
 		strconv.FormatInt(requested, 10), strconv.FormatInt(maxWaitTime.Nanoseconds(), 10),
 		a.maxIdleTimeMillis, a.maxDebtNanos}
 
-	keepTrying := true
 	var waitTime time.Duration
 	var err error
-	for attempt := 0; keepTrying && attempt < a.factory.connectionRetries; attempt++ {
-		client := a.factory.Client().(*redis.Client)
-		res := client.EvalSha(a.factory.scriptSHA, a.keys, args...)
-		switch waitTimeNanos := res.Val().(type) {
-		case int64:
-			waitTime = time.Nanosecond * time.Duration(waitTimeNanos)
-			keepTrying = false
-		default:
-			err = res.Err()
-			if unknownCloseError(err) {
-				logging.Printf("Unknown response '%v' of type %T. Full result %+v",
-					waitTimeNanos, waitTimeNanos, res)
-			}
-
-			a.factory.reconnectToRedis(client)
-		}
-	}
 
-	if keepTrying {
-		// TODO(kaneda): Gracefully handle Redis access errors?
-		logging.Fatalf("Couldn't reconnect to Redis, even after %v attempts with error %+v",
-			a.factory.connectionRetries, err)
+	client := a.factory.Client().(*redis.Client)
+	res := client.EvalSha(a.factory.scriptSHA, a.keys, args...)
+	switch waitTimeNanos := res.Val().(type) {
+	case int64:
+		waitTime = time.Nanosecond * time.Duration(waitTimeNanos)
+		break
+	default:
+		err = res.Err()
+		if unknownCloseError(err) {
+			logging.Printf("Unknown response '%v' of type %T. Full result %+v",
+				waitTimeNanos, waitTimeNanos, res)
+
+		}
+		// Try reconnect to redis
+		a.factory.reconnectToRedis(client)
+		return 0, false
```

Comment:
Reconnect and then return a failure result, regardless whether the reconnect succeeds or fails? I'm not sure I understand the intent here...

Reply:
Yeah it's not perfect. Ideally I want to spin up a go routine and try reconnect in it while failing fast in the `Take` if we're not connected. This will require more work though, and I'd like to have a quick fix in place first.